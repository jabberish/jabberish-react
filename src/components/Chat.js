import React from 'react';
import PropTypes from 'prop-types';

const Chat = ({ messagesData }) => {
  const messages = messagesData.map(message => (
    <li key={message._id}>{message.text}</li>
  ));
  return (
    <section>
      <ul>{messages}</ul>
    </section>
  );
};

Chat.propTypes = {
  messagesData: PropTypes.array.isRequired
};

export default Chat;
