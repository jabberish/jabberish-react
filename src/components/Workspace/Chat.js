import React from 'react';
import PropTypes from 'prop-types';
import MessageInput from './MessageInput';
import Messages from './Messages';

const Chat = ({ messagesData, onSubmitMessage, onUpdateMessageInput, messageInput }) => {
  return (
    <section style={{ width: '100%' }}>
      <Messages messagesData={messagesData} />
      <MessageInput 
        onSubmitMessage={onSubmitMessage} 
        onUpdateMessageInput={onUpdateMessageInput} 
        messageInput={messageInput}
      />
    </section>
  );
};

Chat.propTypes = {
  messagesData: PropTypes.array.isRequired,
  onSubmitMessage: PropTypes.func.isRequired,
  onUpdateMessageInput: PropTypes.func.isRequired,
  messageInput: PropTypes.string.isRequired
};

export default Chat;
