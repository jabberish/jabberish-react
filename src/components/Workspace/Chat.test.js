/* eslint-disable no-console */
import React from 'react';
import { shallow } from 'enzyme';
import Chat from './Chat';

describe('Chat component', () => {
  it('renders Chat', () => {
    const messagesData = [];
    const onSubmitMessage = () => console.log('');
    const onUpdateMessageInput = () => console.log('');
    const messageInput = '';
    const wrapper = shallow(<Chat 
      messagesData={messagesData} 
      onSubmitMessage={onSubmitMessage} 
      onUpdateMessageInput={onUpdateMessageInput}
      messageInput={messageInput} 
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
