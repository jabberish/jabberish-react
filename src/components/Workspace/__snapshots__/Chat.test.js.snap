// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component renders Chat 1`] = `
<section
  style={
    Object {
      "width": "100%",
    }
  }
>
  <Messages
    messagesData={Array []}
  />
  <MessageInput
    messageInput=""
    onSubmitMessage={[Function]}
    onUpdateMessageInput={[Function]}
  />
</section>
`;
